### PR TITLE
Fix padding for IDEFICS 

### LINKED ISSUE #26354: IDEFICS Processor will not respect `padding_side="right"` when `padding="longest"`
### System Info

- `transformers` version: 4.33.2
- Platform: Linux-5.15.0-56-generic-x86_64-with-glibc2.34
- Python version: 3.8.16
- Huggingface_hub version: 0.16.4
- Safetensors version: 0.3.3
- Accelerate version: 0.22.0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

@VictorSanh 

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
from transformers import AutoProcessor

processor = AutoProcessor.from_pretrained("HuggingFaceM4/idefics-9b", padding_side="right")

sents = [["hello world"], [" this is a longer sentence to testing padding"]]

# This is the correct behavior:
a = processor(sents, padding="max_length", truncation=True, max_length=20)
print(processor.tokenizer.decode(a['input_ids'][0]))
# => <s> hello world<unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk>

# This is the incorrect behavior:
b = processor(sents, padding="longest", truncation=True, max_length=30)

print(processor.tokenizer.decode(b['input_ids'][0]))
# => <unk><unk><unk><unk><unk><unk><s> hello world
```

### Expected behavior

It should pad on right side

### PR DESCRIPTION
**Pull request by shauray8**
# What does this PR do?

Fixes padding issues - 
```python
from transformers import AutoProcessor

processor = AutoProcessor.from_pretrained("HuggingFaceM4/idefics-9b", padding_side="right")

sents = [["hello world"], [" this is a longer sentence to testing padding"]]

# This is the correct behaviour:
a = processor(sents, padding="max_length", truncation=True, max_length=20)
print(processor.tokenizer.decode(a['input_ids'][0]))
# => <s> hello world<unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk>

# This is the incorrect behaviour:
b = processor(sents, padding="longest", truncation=True, max_length=30)

print(processor.tokenizer.decode(b['input_ids'][0]))
# => <unk><unk><unk><unk><unk><unk><s> hello world
```
With this fix the results would be 
```
# for max_length
<s> hello world<unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk><unk>
# for longest
<s> hello world<unk><unk><unk><unk><unk><unk>
```

Fixes #26354

## Before submitting
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.

## Who can review?
@VictorSanh @ArthurZucker @LysandreJik 

*let me know if there's anything that needs follow-ups*

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
Great to see a detailed issue and detailed PR! Thanks for the effort 😉 
Can you add a test (consisting of your snipper for example) in the `test_processing_idefics.py` file?

**Comment by shauray8** [comment]
sure!

**Comment by shauray8** [comment]
@ArthurZucker CI looks green, PR's ready for another review.

**Review by ArthurZucker — APPROVED** [review]
Thanks for the fix 😉
